packages/react-sketch-canvas/src/Canvas/index.tsx

```diff
@@ -291,13 +311,31 @@ release drawing even when point goes out of canvas */
     );
   }, [paths]);
 
+  // avoid pen from scrolling if scrollOnTouch
+  useEffect(() => {
+    const listener = function(
+      this: HTMLDivElement,
+      event: TouchEvent
+    ): void {
+      if (currentPointerType.current === "pen") {
+        event.preventDefault();
+        event.stopPropagation();
+      }
+    };
+
+    if (scrollOnTouch) {
+      canvasRef.current?.addEventListener("touchstart", listener, { passive: false });
+    }
+    return () => canvasRef.current?.removeEventListener("touchstart", listener);
```

Comment:
`canvasRef.current` might get cleaned up before this cleanup is run. so `canvasRef.current` must be copied to a variable in the scope of this function. and then you can add or remove event listeners.